src/hooks/usePoolData.ts

```diff
@@ -101,28 +93,22 @@ const emptyPoolData = {
   isPaused: false,
   isMigrated: false,
   sdlPerDay: null,
+  isGuarded: false,
+  isMetaSwap: false,
+  poolType: PoolTypes.OTHER,
 } as PoolDataType
 
 export default function usePoolData(
-  poolName?: PoolName,
+  poolName?: PoolName | string,
 ): PoolDataHookReturnType {
   const { account, library, chainId } = useActiveWeb3React()
-  const swapContract = useSwapContract(poolName)
-  const rewardsContract = useMiniChefContract()
-  const migratorContract = useGeneralizedSwapMigratorContract()
-  const { tokenPricesUSD, lastTransactionTimes, swapStats } = useSelector(
+  const basicPools = useContext(BasicPoolsContext)
+  const tokens = useContext(TokensContext)
+  const userState = useContext(UserStateContext)
+  const minichefData = useContext(MinichefContext)
```

Comment:
https://github.com/dai-shi/use-context-selector#usage

might want to bring this in, in the short term as a bandaid. I'll work on a react-query based solution in the meantime.

Reply:
This seems like redux with extra steps 🤨

Reply:
It's just calling useContextSelector instead of useContext to enable context scoping to prevent re-renders. Either way there are better patterns than this and better ways to manage server-state than Redux.